smokefree/sites/all/modules/custom/icfi_api_keys/ICFIApi.inc

```diff
@@ -0,0 +1,104 @@
+<?php
+
+
+class ICFIApi {
+  protected $machineName;
+  protected $apiValue;
+  protected $isJsVariable;
+
+  public function __construct($machineName, $apiValue, $isJsVariable) {
+    if (empty($machineName) || empty($apiValue)) {
+      throw new InvalidArgumentException('Please supply a valid api value and an appropriate machine name');
```

Comment:
If $isJsVariable can be none, should it have a default value?

Reply:
That is handled in the .install file. The default value is 0